src/x11/request/graphics.rs

```diff
@@ -2102,7 +2102,18 @@ impl Writable for DrawText8 {
 		buf.put_u8(Self::MAJOR_OPCODE);
 		// Unused metabyte position.
 		buf.put_u8(0);
+		#[cfg(not(feature = "big-requests"))]
 		buf.put_u16(self.length());
+		#[cfg(feature = "big-requests")]
+		{
+			let length = self.length();
+			if let Ok(length) = length.try_into() {
+				buf.put_u16(length);
+			} else {
+				buf.put_u16(0);
+				buf.put_u32(length);
+			}
+		}
```

Comment:
Maybe this would be better as a separate function somewhere?  Also, doesn't matter much but might be clearer if formatted like this:

```suggestion
		{
			let length = self.length();

			match u16::try_from(length) {
				Ok(length) => buf.put_u16(length),

				// length is too big for u16
				Err(_) => {
					buf.put_u16(0);
					buf.put_u32(length);
				},
			}
		}
```